validphys2/src/validphys/n3fit_data_utils.py

```diff
@@ -94,7 +95,12 @@ def validphys_group_extractor(datasets, tr_masks):
     # Use zip_longest since tr_mask can be (and it is fine) an empty list
     for dspec, mask in zip_longest(datasets, tr_masks):
         # Load all fktables with the appropiate cuts
-        fktables = [fk.load_with_cuts(dspec.cuts) for fk in dspec.fkspecs]
+        fktables = [load_cached_fk_tables(fk, dspec.cuts) for fk in dspec.fkspecs]
         # And now put them in a FittableDataSet object which
         loaded_obs.append(FittableDataSet(dspec.name, fktables, dspec.op, dspec.frac, mask))
     return loaded_obs
+
+@functools.lru_cache
+def load_cached_fk_tables(fk, cuts):
+    return fk.load_with_cuts(cuts)
```

Comment:
Wouldn't it be better to wrap directly `fk.load_with_cuts`?

https://github.com/NNPDF/nnpdf/blob/f02b49a8e6eb785af6a56dc3195133abff571046/validphys2/src/validphys/core.py#LL442C4-L442C4

(in practical terms it should be the same)

Given an fktable and a set of cuts, I see no reason why we would we want two different objects so that probably is a positive change in other parts of the code as well*


*hopefully

Reply:
Yes that would be the more elegant solution, but needs thorough testing because this indeed impacts the code in more places, there will be shared objects where there were copies before. I can have a look.